src/ApplicationInsights.Kubernetes/TelemetryInitializers/KubernetesTelemetryInitializer.cs

```diff
@@ -142,6 +142,7 @@ private void SetCustomDimensions(ISupportProperties telemetry)
             SetCustomDimension(telemetry, Invariant($"{K8s}.{Pod}.ID"), this._k8sEnvironment.PodID);
             SetCustomDimension(telemetry, Invariant($"{K8s}.{Pod}.Name"), this._k8sEnvironment.PodName);
             SetCustomDimension(telemetry, Invariant($"{K8s}.{Pod}.Labels"), this._k8sEnvironment.PodLabels);
+            SetCustomDimension(telemetry, Invariant($"{K8s}.{Pod}.Namespace"), this._k8sEnvironment.PodNamespace, isValueOptional: true);
```

Comment:
Question unrelated to the change: does calling `Invariant($"something something")` incurs formatting and `ToString()` cost for every piece of telemetry that is augmented by `KubernetesInitializer`?

Reply:
Thanks for your review. That's an interesting question. I have created #206 to discuss this topic.